### DISCUSSION TITLE
32-bit CLAP plugins?

### DISCUSSION THREAD

**Discussion by JBetz** [free-audio/clap#178] [Q&A] [answered]
I'm trying to load the clap-saw-demo plugin into Dolphin Smalltalk and getting "X is not a valid Win32 application error". Dolphin is 32-bit so I'm assuming that the plugin is 64-bit since it doesn't throw that error when loading 32-bit VSTs.

Are there any CLAP plugins that can be built to run 32-bit? Or is 64-bit the de-facto standard?

**Comment by baconpaul** [comment]
I don't know if anyone has built any 32 bit claps. There's a small chance I forgot to turn them off on the surge xt builds though so I would start with the 32 bit surge xt 1.1. installer.

clap-saw-demo should compile 32 bit if you add a "-A Win32" to the cmake, perhaps, but i've certainly never tested it.

But to answer your question: I don't think anyone is going to support 32 bit clap on windows with any vigor no. So if you use it in this fashion my guess is your available synths will be open source ones you build yourself more than the commercial lines. While it is certainly not a requirement that you are 64 bit, I think everyone just kinda is.

**Comment by baconpaul** [comment, marked as answer]
Yeah I just checked.

https://surge-synthesizer.github.io/downloads

the surge 32 bit installer has a clap included which is built by our pipelines

I don't think anyone anywhere in the world has ever run it though, since as far as I know, until your question here, there haven't been any 32 bit windows hosts.

If it works. though, that would be cool!

**Reply by sagantech** [reply]
I will run it. I want to test a 32 bit clap plug-in for Windows and maybe Mac. My host compiles as 32/64 bit.

**Reply by sagantech** [reply]
Are you saying that "⇩ [Windows 32-bit (Standalone, VST3)](https://github.com/surge-synthesizer/releases-xt/releases/download/1.1.1/surge-xt-win32-1.1.1-setup.exe)" includes a 32 bit .clap plugin?

**Reply by baconpaul** [reply]
I'm saying that I think it does, and looking at the build logs it seems it would, even though the documentation is wrong, but the last time I tested win 32 bit was when i was fixing the sse2 alignment problems 2.5 years ago so it just sort of plugs along in CI.

for surge at least, zero chance it will compile mac 32 bit. We doo too much arm cross compiling in the build stage for lua.

**Reply by baconpaul** [reply]
by the way if it does we should (of course) fix up that link to include CLAP :) I think we probably left it out because it wasn't there, then we added it, but didn't fix the web.

**Reply by baconpaul** [reply]
And appreciate the test! Curious if it works :)

**Reply by sagantech** [reply]
32 bit clap is in the installer. I installed it and it works beautifully from my initial testing.

**Reply by sagantech** [reply]
Also the FX plug is there. Will try that one as well later.

**Reply by sagantech** [reply]
Got a random crash on playback after recording so let me do some more research.

**Reply by sagantech** [reply]
So I retried and there was no crash then I went and installed the 64 bit (CLAP) version and to my amazement, I went and opened the document I had saved using the Surge-XT 32 bit and Surge-XT  64 bit opened and played back perfectly. Nice job.

**Reply by sagantech** [reply]
@baconpaul Can you do me a favor and add "synthesizer" to your clap_plugin_descriptor.features array. Metro sets the category of the plug-in automatically if it finds a matching category (instrument is not a recognized category)?

**Reply by baconpaul** [reply]
I can but instrument is in the set fyi 

https://github.com/free-audio/clap/blob/af1e05ce6487151c6b36003fdb47566dbb50194e/include/clap/plugin-features.h#L18

**Reply by sagantech** [reply]
I know but "instrument" is not a category **I** recognize and may be an "instrument model". "Synthesizer", which I notice is already defined as CLAP_PLUGIN_FEATURE_SYNTHESIZER  appears to be missing from the features array in the CLAP version.

**Reply by sagantech** [reply]
I guess CLAP calls it a "sub-category".

**Reply by baconpaul** [reply]
Yeah I will add it for sure when I fix the vel0 bug. Just letting you know that there may be quite a few plugins which just say instrument out there

**Reply by sagantech** [reply]
First, this is a minor issue. The user can easily set the category herself. Second, the documentation, in features.h, for instrument says: 

> // Add this feature if your plugin can process note events and then produce audio

This could easily be an effect of some kind as well since it says nothing about audio in.

**Reply by baconpaul** [reply]
Yup! Surge is an instrument and an effect both for instance since the synth advertises a side chain in. But labeling it an effect would be obnoxious albeit correct. Genres and tags are always difficult though

but definitely surge is a shynthesizer and it’s obviousky an omission that it doesn’t say that :)
